modules/api/pkg/ee/clusterbackup/storage-location/provider.go

```diff
@@ -26,77 +26,108 @@ package storagelocation
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	apiv2 "k8c.io/dashboard/v2/pkg/api/v2"
+	"k8c.io/dashboard/v2/pkg/handler/v1/common"
 	"k8c.io/dashboard/v2/pkg/provider"
+	"k8c.io/dashboard/v2/pkg/provider/kubernetes"
 	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
 	"k8c.io/kubermatic/v2/pkg/resources"
+	utilerrors "k8c.io/kubermatic/v2/pkg/util/errors"
 
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apiserver/pkg/storage/names"
+	restclient "k8s.io/client-go/rest"
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type BackupStorageProvider struct {
-	privilegedClient ctrlruntimeclient.Client
+	createMasterImpersonatedClient kubernetes.ImpersonationClient
+	privilegedClient               ctrlruntimeclient.Client
 }
 
 var _ provider.BackupStorageProvider = &BackupStorageProvider{}
 
-func NewBackupStorageProvider(privilegedClient ctrlruntimeclient.Client) *BackupStorageProvider {
+const (
+	ownerReferenceApi  = "kubermatic.k8s.io/v1"
+	ownerReferenceKind = "ClusterBackupStorageLocation"
+)
+
+func NewBackupStorageProvider(createMasterImpersonatedClient kubernetes.ImpersonationClient, privilegedClient ctrlruntimeclient.Client) *BackupStorageProvider {
 	return &BackupStorageProvider{
-		privilegedClient: privilegedClient,
+		createMasterImpersonatedClient: createMasterImpersonatedClient,
+		privilegedClient:               privilegedClient,
 	}
 }
 
-func (p *BackupStorageProvider) ListUnsecured(ctx context.Context, labelSet map[string]string) (*kubermaticv1.ClusterBackupStorageLocationList, error) {
+func (p *BackupStorageProvider) List(ctx context.Context, userInfo *provider.UserInfo, labelSet map[string]string) (*kubermaticv1.ClusterBackupStorageLocationList, error) {
+	if userInfo == nil {
+		return nil, errors.New("a user is missing but required")
+	}
+
 	listOpts := &ctrlruntimeclient.ListOptions{
 		LabelSelector: labels.SelectorFromSet(labelSet),
 		Namespace:     resources.KubermaticNamespace,
 	}
 
 	cbslList := &kubermaticv1.ClusterBackupStorageLocationList{}
-	if err := p.privilegedClient.List(ctx, cbslList, listOpts); err != nil {
+	client := p.privilegedClient
+
+	if err := client.List(ctx, cbslList, listOpts); err != nil {
```

Comment:
nit: just to keep this uniform with other code. Let's go back to the previous p.privilegedClient.List notation from before